<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Differential analysis of HiChIP loops &#8212; FitHiChIP 0.1 documentation</title>
    <link rel="stylesheet" href="../static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="../static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../static/documentation_options.js"></script>
    <script type="text/javascript" src="../static/jquery.js"></script>
    <script type="text/javascript" src="../static/underscore.js"></script>
    <script type="text/javascript" src="../static/doctools.js"></script>
    <script type="text/javascript" src="../static/language_data.js"></script>
    <script type="text/javascript" src="../static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="../static/bootstrap-3.3.7/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="../static/bootstrap-sphinx.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Various utility scripts / functions" href="Utilities.html" />
    <link rel="prev" title="Details of FitHiChIP outputs" href="output.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head><body>

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../index.html">
          FitHiChIP</a>
        <span class="navbar-text navbar-version pull-left"><b></b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick start</a></li>
<li class="toctree-l1"><a class="reference internal" href="configuration.html">Setting up configuration file</a></li>
<li class="toctree-l1"><a class="reference internal" href="output.html">Details of FitHiChIP outputs</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Differential analysis of HiChIP loops</a></li>
<li class="toctree-l1"><a class="reference internal" href="Utilities.html">Various utility scripts / functions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked question</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Differential analysis of HiChIP loops</a><ul>
<li><a class="reference internal" href="#example-1">Example 1</a></li>
<li><a class="reference internal" href="#example-2">Example 2</a></li>
<li><a class="reference internal" href="#details-of-differential-analysis-outputs">Details of differential analysis outputs</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="output.html" title="Previous Chapter: Details of FitHiChIP outputs"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Details of Fi...</span>
    </a>
  </li>
  <li>
    <a href="Utilities.html" title="Next Chapter: Various utility scripts / functions"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Various utili... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="../sources/usage/DiffLoops.rst.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12 content">
      
  <div class="section" id="differential-analysis-of-hichip-loops">
<h1>Differential analysis of HiChIP loops<a class="headerlink" href="#differential-analysis-of-hichip-loops" title="Permalink to this headline">¶</a></h1>
<p>FitHiChIP is used to find out the differential HiChIP loops between two categories, each with multiple replicates.</p>
<p>The script ** Differetial_Analysis_Script.sh **  contains command to execute differential analysis.</p>
<p>Source code of the differential analysis is provided in <strong>Imp_Scripts/DiffAnalysisHiChIP.r</strong></p>
<p>Parameters associated with this script is:</p>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">--AllLoopList</span></kbd></td>
<td>Comma or colon separated list of loops of all categories and all replicates, with FitHiChIP significance (FDR) values.
User should provide the file <strong>*PREFIX*.interactions_FitHiC.bed</strong>
(please check the page <a class="reference internal" href="output.html#details-of-fithichip-outputs"><span class="std std-ref">Details of FitHiChIP outputs</span></a>) which contains all loops for a specific sample
along with FitHiChIP generated significance values. Mandatory parameter.</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;ChrSizeFile           File having the chromosome sizes for the reference genome. For hg19 reference genome, user can use the file
<em>chrom_hg19.sizes</em> which is already provided within the TestData folder. Mandatory parameter.</p>
</div></blockquote>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">--FDRThr</span></kbd></td>
<td>FDR significance threshold for FitHiChIP loops. Default = 0.01</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;CovThr                        Maximum difference of ChIP-seq coverage (in percentage) between two categories to label a non-differential (by EdgeR) 1D bin as
ND (i.e. no difference). Default is 25, means non-differential 1D bins with difference of ChIP coverage &lt; 25% are considered as ND.</p>
</div></blockquote>
<blockquote>
<div><dl class="docutils">
<dt>–ChIPAlignFileList Comma or colon separated list of ChIP-seq alignment files. User can 1) either provide two files, one for each category,</dt>
<dd><ol class="first last arabic simple" start="2">
<li>or provide ChIP seq alignment files one for each sample. Mandatory parameter.</li>
</ol>
</dd>
</dl>
<p class="attribution">&mdash;OutDir                        Output directory to contain differential analysis results. Mandatory parameter.</p>
</div></blockquote>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">--CategoryList</span></kbd></td>
<td>Comma or colon separated list of strings, representing labels of two categories. Default: Category1, Category2.</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;ReplicaCount          Comma or colon separated list representing the count of samples (replicates) belonging to individual categories.
Default: 1,1 (means that we are considering one replicate per sample).</p>
</div></blockquote>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--ReplicaLabels1</span></kbd></td>
</tr>
<tr><td>&#160;</td><td>Comma or colon separated list of the label of replicates for the first category. Default: R1,R2, etc.</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;ReplicaLabels2        Comma or colon separated list of the label of replicates for the second category. Default: R1,R2, etc.</p>
</div></blockquote>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group" colspan="2">
<kbd><span class="option">--FoldChangeThr</span></kbd></td>
</tr>
<tr><td>&#160;</td><td>EdgeR fold change threshold. Default = 2 means log2 fold change threshold of 1 is applied on the EdgeR output.</td></tr>
</tbody>
</table>
<p class="attribution">&mdash;DiffFDRThr            FDR threshold for EdgeR. Default is 0.05, means that loops with FDR &lt; 0.05, and fold change &gt;= log2(FoldChangeThr)
would be considered as differential.</p>
</div></blockquote>
<blockquote>
<div><table class="docutils option-list" frame="void" rules="none">
<col class="option" />
<col class="description" />
<tbody valign="top">
<tr><td class="option-group">
<kbd><span class="option">--bcv</span></kbd></td>
<td>If number of replicates is 1 for either of the input categories, this value is the square-root-dispersion.
Default and recommended value is 0.4. For details, please see the EdgeR manual.</td></tr>
</tbody>
</table>
</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">File specified in the parameter –AllLoopList can be in gzipped format as well.</p>
</div>
<div class="section" id="example-1">
<h2>Example 1<a class="headerlink" href="#example-1" title="Permalink to this headline">¶</a></h2>
<p>Suppose, we have two input categories denoted as <em>cat1</em> and <em>cat2</em>. Each category has two replicates, denoted as <em>repl1</em> and <em>repl2</em>.
Assume hg19 as the reference genome, and we have one ChIP-seq alignment file for individual categories. Then, the command for differential analysis would be:</p>
<blockquote>
<div><p>Rscript Imp_Scripts/DiffAnalysisHiChIP.r –AllLoopList cat1_repl1_file,cat1_repl2_file,cat2_repl1_File,cat2_repl2_file –ChrSizeFile TestData/chrom_hg19.sizes –FDRThr 0.01 –CovThr 25 –ChIPAlignFileList cat1_ChIPAlign.bam,cat2_ChIPAlign.bam –OutDir /home/user/diffanalysis/outdir –CategoryList “CellLine1”,”CellLine2” –ReplicaCount 2,2 –ReplicaLabels1 “R1”,”R2” –ReplicaLabels2 “R1”,”R2” –FoldChangeThr 2 –DiffFDRThr 0.05 –bcv 0.4</p>
<ul class="simple">
<li>Here, <em>cat1_repl1_file</em> denotes the loops of category 1, replicate 1, containing FitHiChIP significance values. Similar notations for the other files.</li>
<li>In the parameter –AllLoopList, all the samples of category 1 should be specified first, followed by the samples in category 2</li>
<li>User can also specify the files using colons (:) as a separator instead of comma.</li>
<li>The parameter –FDRThr corresponds to the significance threshold of FDR employed for FitHiChIP loops.</li>
<li>The file <em>cat1_ChIPAlign.bam</em> represents ChIP-seq alignment file for the category 1. Similar notation for category 2.</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="example-2">
<h2>Example 2<a class="headerlink" href="#example-2" title="Permalink to this headline">¶</a></h2>
<p>Suppose, we have two input categories denoted as <em>cat1</em> and <em>cat2</em>. Each category has two replicates, denoted as <em>repl1</em> and <em>repl2</em>.
Assume hg19 as the reference genome, and we have one ChIP-seq alignment file for individual samples. Then, the command for differential analysis would be:</p>
<blockquote>
<div><p>Rscript Imp_Scripts/DiffAnalysisHiChIP.r –AllLoopList cat1_repl1_file,cat1_repl2_file,cat2_repl1_File,cat2_repl2_file –ChrSizeFile TestData/chrom_hg19.sizes –FDRThr 0.01 –CovThr 25 –ChIPAlignFileList cat1_R1_ChIPAlign.bam,cat1_R2_ChIPAlign.bam,cat2_R1_ChIPAlign.bam,cat2_R2_ChIPAlign.bam –OutDir /home/user/diffanalysis/outdir –CategoryList “CellLine1”,”CellLine2” –ReplicaCount 2,2 –ReplicaLabels1 “R1”,”R2” –ReplicaLabels2 “R1”,”R2” –FoldChangeThr 2 –DiffFDRThr 0.05 –bcv 0.4</p>
<ul class="simple">
<li>Here, <em>cat1_R1_ChIPAlign.bam</em> denotes the ChIP-seq alignment for replicate 1 of the category 1. Similar notations for the other files.</li>
</ul>
</div></blockquote>
</div>
<div class="section" id="details-of-differential-analysis-outputs">
<h2>Details of differential analysis outputs<a class="headerlink" href="#details-of-differential-analysis-outputs" title="Permalink to this headline">¶</a></h2>
<p>Here we describe important files and folders generated by the differential analysis, and placed within the specified output directory <strong>OutDir</strong>.</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first"><strong>ChIP_Coverage_ALL.bed</strong> : lists the ChIP-seq coverage of fixed size bins for both input categories.</p>
</li>
<li><p class="first"><strong>Input_Parameters*.log</strong> : lists the input command line options.</p>
</li>
<li><dl class="first docutils">
<dt><strong>MasterSheet_*Loops.bed</strong> <span class="classifier-delimiter">:</span> <span class="classifier">Union of all FitHiChIP loops of the given categories and samples,</span></dt>
<dd><p class="first last">with the respective contact counts and interaction significance values.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><strong>EdgeR_Loops_Ov_FitHiChIP_Sig_One_Repl</strong> <span class="classifier-delimiter">:</span> <span class="classifier">Folder containing the EdgeR differential loops which are significant (in terms of FitHiChIP FDR threshold)</span></dt>
<dd><blockquote class="first">
<div><p>in at least one input sample.</p>
</div></blockquote>
<ul class="simple">
<li>Underlying folders have name <strong>X-Y</strong> where <strong>X</strong> and <strong>Y</strong> denote the type of 1D bin (LD, ND or HD) according to their ChIP-coverage specific</li>
</ul>
<p>differential analysis.</p>
<ul class="last simple">
<li>Within each folder, file <strong>DiffLoops.bed</strong> contains differential loops of the particular category.</li>
<li>Files <strong>DiffLoops_Excl_*.bed</strong> contain differential loops exclusive to one category (i.e. significant in only that category).</li>
</ul>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2019, Sourya Bhattacharyya, Ferhat Ay.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.3.<br/>
    </p>
  </div>
</footer>
  </body>
</html>