<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Details of FitHiChIP outputs &#8212; FitHiChIP 0.1 documentation</title>
    <link rel="stylesheet" href="../static/bootstrap-sphinx.css" type="text/css" />
    <link rel="stylesheet" href="../static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../static/documentation_options.js"></script>
    <script type="text/javascript" src="../static/jquery.js"></script>
    <script type="text/javascript" src="../static/underscore.js"></script>
    <script type="text/javascript" src="../static/doctools.js"></script>
    <script type="text/javascript" src="../static/language_data.js"></script>
    <script type="text/javascript" src="../static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="../static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="../static/bootstrap-3.3.7/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="../static/bootstrap-sphinx.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Differential analysis of HiChIP loops" href="DiffLoops.html" />
    <link rel="prev" title="Setting up configuration file" href="configuration.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head><body>

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="../index.html">
          FitHiChIP</a>
        <span class="navbar-text navbar-version pull-left"><b></b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="../index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick start</a></li>
<li class="toctree-l1"><a class="reference internal" href="configuration.html">Setting up configuration file</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Details of FitHiChIP outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffLoops.html">Differential analysis of HiChIP loops</a></li>
<li class="toctree-l1"><a class="reference internal" href="Utilities.html">Various utility scripts / functions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked question</a></li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Details of FitHiChIP outputs</a><ul>
<li><a class="reference internal" href="#summary-html-file">Summary HTML file</a></li>
<li><a class="reference internal" href="#details-of-output-files">Details of output files</a></li>
<li><a class="reference internal" href="#visualizing-significant-interactions-in-washu-epigenome-browser">Visualizing significant interactions in WashU epigenome browser</a><ul>
<li><a class="reference internal" href="#using-current-version">Using <strong>current</strong> version</a></li>
<li><a class="reference internal" href="#using-legacy-version">Using <strong>legacy</strong> version</a></li>
</ul>
</li>
<li><a class="reference internal" href="#visualizing-significant-interactions-in-ucsc-genome-browser">Visualizing significant interactions in UCSC genome browser</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="configuration.html" title="Previous Chapter: Setting up configuration file"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Setting up co...</span>
    </a>
  </li>
  <li>
    <a href="DiffLoops.html" title="Next Chapter: Differential analysis of HiChIP loops"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">Differential ... &raquo;</span>
    </a>
  </li>
              
            
            
            
            
              <li class="hidden-sm">
<div id="sourcelink">
  <a href="../sources/usage/output.rst.txt"
     rel="nofollow">Source</a>
</div></li>
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="../search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
    <div class="col-md-12 content">
      
  <div class="section" id="details-of-fithichip-outputs">
<h1>Details of FitHiChIP outputs<a class="headerlink" href="#details-of-fithichip-outputs" title="Permalink to this headline">¶</a></h1>
<div class="section" id="summary-html-file">
<h2>Summary HTML file<a class="headerlink" href="#summary-html-file" title="Permalink to this headline">¶</a></h2>
<p>Suppose <strong>OutDir</strong> denotes the specified output directory in the configuration file.</p>
<p>Then, successful execution of FitHiChIP pipeline generates an HTML file <strong>Summary_results_FitHiChIP.html</strong> within this output directory <strong>OutDir</strong>.</p>
<p>This file lists all output files generated from the FitHiChIP pipeline, according to the given input parameters.</p>
</div>
<div class="section" id="details-of-output-files">
<h2>Details of output files<a class="headerlink" href="#details-of-output-files" title="Permalink to this headline">¶</a></h2>
<p>Here we describe all the files and folders existing within the specified output directory <strong>OutDir</strong>.</p>
<p>User may additionally open this page <a class="reference internal" href="configuration.html#setting-up-configuration-file"><span class="std std-ref">Setting up configuration file</span></a> for checking the configuration parameters and analyzing the below mentioned output files and folders:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first"><strong>Parameters.txt</strong>: file listing the parameters used.</p>
</li>
<li><p class="first"><strong>HiCPro_Matrix_BinSize”BINSIZE”</strong>:</p>
<blockquote>
<div><p>If the parameters (A.2) and (A.3) are empty, this folder contains the interaction matrix files generated from the input valid pairs file.</p>
<p>The interaction matrix files are <em>MatrixHiCPro_abs.bed</em> and <em>MatrixHiCPro.matrix</em>.</p>
<ul>
<li><p class="first"><strong>*.interactions.initial.bed</strong>: CIS locus pairs along with their contact counts, corresponding to the input valid pairs (or contact matrix).</p>
</li>
<li><p class="first"><strong>L_”l”_U”u”</strong>: folder name according to the distance thresholds “l” and “u” specified in the parameters (B.7) and (B.8).</p>
<blockquote>
<div><ul class="simple">
<li><strong>*.cis.interactions.DistThr.bed</strong>: File containing CIS interactions within the distance range from “l” to “u”.</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ol>
<ol class="arabic" start="4">
<li><p class="first"><strong>NormFeatures</strong>:</p>
<blockquote>
<div><p>Folder containing various statistical features computed for individual genomic intervals / bins (with respect to the specified bin size).</p>
<ul>
<li><p class="first">User should specifically look at the file <strong>*.AllBin_CompleteFeat.bed</strong>, having the following format:</p>
<blockquote>
<div><ul class="simple">
<li>Columns 1 to 3: chromosome bin.</li>
<li>Column 4: Coverage of this bin (number of reads mapped onto it)</li>
<li>Column 5: if 1, this bin is a <em>peak-bin</em> (overlapping with the reference peaks, provided in the parameter B.1); otherwise (0) <em>non-peak bin</em>.</li>
<li>Column 6: Bias value of a bin; either coverage or ICE bias depending on the parameter B.11</li>
<li>Columns 7 and 8: unused columns.</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><strong>FitHiChIP_”INTTYPE”_b”BINSIZE”_L”l”_U”u”</strong></p>
<blockquote>
<div><p>Folder according to the specified foreground (type of interactions to be reported) as per the parameter B.5, bin size (parameter B.6),
and the distance thresholds (parameters B.7 and B.8)</p>
<ul class="simple">
<li>INTTYPE is a string with values ALL2ALL (when the parameter (B.5) = 4), Peak2Peak (parameter (B.5) = 1), Peak2NonPeak (parameter (B.5) = 2), or</li>
</ul>
<p>Peak2ALL (parameter (B.5) = 3).</p>
<ul>
<li><p class="first">Within this folder, a directory structure of the following name (pattern) is created:</p>
<blockquote>
<div><p><strong>P2PBckgr_[UseP2PBackgrnd]/[BiasType]</strong></p>
</div></blockquote>
</li>
</ul>
<p>Where:</p>
<blockquote>
<div><ul class="simple">
<li>[UseP2PBackgrnd]: Value of the parameter (B.10)</li>
<li>[BiasType]: A string, either “Coverage_Bias” or “ICE_Bias”, according to the parameter (B.11).</li>
</ul>
</div></blockquote>
<ul>
<li><p class="first">Within this directory, following files are important:</p>
<blockquote>
<div><ul class="simple">
<li><strong>Interactions.sortedGenDist.bed</strong>: Interactions sorted by increasing genomic distance. Used as the input to FitHiChIP.</li>
<li><strong>FitHiC_BiasCorr</strong>:</li>
</ul>
<p>Within this folder, following files and folders reside:</p>
<blockquote>
<div><ul>
<li><p class="first"><strong>Bin_Info.log</strong>: Equal occupancy bins employed in FitHiChIP</p>
</li>
<li><p class="first"><strong>configuration.txt</strong>: list of configuration parameters for FitHiChIP</p>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC.bed</strong>: Lists all interactions along with their significance (p-value and FDR) computed using FitHiChIP.</p>
<blockquote>
<div><div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">For differential analysis of HiChIP loops (page <a class="reference internal" href="DiffLoops.html#differential-analysis-of-hichip-loops"><span class="std std-ref">Differential analysis of HiChIP loops</span></a>), this file is to be provided as input.</p>
</div>
</div></blockquote>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”.bed</strong>:</p>
<blockquote>
<div><p>Significant interactions according to the q-value threshold (qval) specified in parameter (B.9).</p>
<p>For example, if qval=0.01 (default), the file name becomes <strong>*PREFIX*.interactions_FitHiC_Q0.01.bed</strong>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic">
<li><p class="first">This file is the set of significant interactions generated by FitHiChIP, without applying any merge filtering.</p>
</li>
<li><p class="first">For peak-to-all foreground (default output mode), if <strong>UseP2PBackgrnd</strong> = 0, then this set of loops refer to FitHiChIP(L).</p>
</li>
<li><p class="first">If <strong>UseP2PBackgrnd</strong> = 1, this set of loops refer to FitHiChIP(S) output.</p>
</li>
<li><p class="first">For a custom FDR threshold (say 0.05), user may use the following awk script to generate the significant interactions.</p>
<blockquote>
<div><p>awk ‘$NF&lt;0.05’ <em>PREFIX</em>.interactions_FitHiC.bed &gt; FitHiChIP_out_Q0.05.bed</p>
</div></blockquote>
</li>
</ol>
</div>
</div></blockquote>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”_WashU.bed.*</strong>:</p>
<blockquote>
<div><p>Significant FitHiChIP loops converted to WashU epigenome browser compatible format.</p>
</div></blockquote>
</li>
<li><p class="first"><strong>Merge_Nearby_Interactions</strong>:</p>
<blockquote>
<div><p>Directory containing the output of merge filtering operation, if the parameter (B.12) = 1.</p>
<ul>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”_MergeNearContacts.bed</strong>:</p>
<blockquote>
<div><p>FitHiChIP significant loops returned after the merge filtering operation.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="last arabic simple">
<li>For peak-to-all foreground (default output mode), if <strong>UseP2PBackgrnd</strong> = 0, this set of loops refer to FitHiChIP(L+M) output.</li>
<li>If <strong>UseP2PBackgrnd</strong> = 1, this set of loops are denoted as FitHiChIP(S+M) in the manuscript.</li>
</ol>
</div>
</div></blockquote>
</li>
<li><p class="first"><strong>*PREFIX*.interactions_FitHiC_Q”qval”_MergeNearContacts_WashU.bed.*</strong>:</p>
<blockquote>
<div><p>WashU epigenome browser compatible loops.</p>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="visualizing-significant-interactions-in-washu-epigenome-browser">
<h2>Visualizing significant interactions in WashU epigenome browser<a class="headerlink" href="#visualizing-significant-interactions-in-washu-epigenome-browser" title="Permalink to this headline">¶</a></h2>
<p>FitHiChIP interactions can be loaded in the WashU epigenome browser for visualizing the significant interactions.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>Two different versions of WashU epigenome browser currently exist:</p>
<blockquote>
<div><p><strong>current</strong> (&lt;<a class="reference external" href="http://epigenomegateway.wustl.edu/browser/">http://epigenomegateway.wustl.edu/browser/</a>&gt;) and</p>
<p><strong>legacy</strong> (&lt;<a class="reference external" href="http://epigenomegateway.wustl.edu/legacy/">http://epigenomegateway.wustl.edu/legacy/</a>&gt;).</p>
</div></blockquote>
<p class="last">These two versions have substantial differences, so we describe using both of these versions.</p>
</div>
<div class="section" id="using-current-version">
<h3>Using <strong>current</strong> version<a class="headerlink" href="#using-current-version" title="Permalink to this headline">¶</a></h3>
<p>The files <strong>*_WashU.bed.gz</strong> mentioned above (FitHiChIP generated significant loops, with or without applying merged filtering) are all
compatible with the <strong>current</strong> version. User can visualize these tracks by following these steps:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Open &lt;<a class="reference external" href="http://epigenomegateway.wustl.edu/browser/">http://epigenomegateway.wustl.edu/browser/</a>&gt;</li>
<li>Select reference genome.</li>
<li>select <strong>Tracks</strong> -&gt; <strong>upload Local Track</strong></li>
<li>Choose track file type: <strong>longrange</strong></li>
<li>Choose track file: select both <strong>*_WashU.bed.gz</strong> and <strong>*_WashU.bed.gz.tbi</strong> files <strong>together (2 files)</strong>.</li>
<li>Close the track upload box (right corner - (X) sign)</li>
<li>Now right click on the track name, and select <strong>DISPLAY MODE</strong> as <strong>ARC</strong></li>
</ol>
</div></blockquote>
</div>
<div class="section" id="using-legacy-version">
<h3>Using <strong>legacy</strong> version<a class="headerlink" href="#using-legacy-version" title="Permalink to this headline">¶</a></h3>
<p>User would need to convert the FitHiChIP significant loops (in .bed format, not gzipped version) as the following:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Output of FitHiChIP(L) or FitHiChIP(S) - file <strong>FitHiC_Q”qval”.bed</strong></p>
<blockquote>
<div><p>Let us denote the file as <em>inp.bed</em>.
It contains header line, first 6 fields denote the interacting bins and the last field denotes the q-value.
User needs to apply the following awk script to this file:</p>
<blockquote>
<div><p>awk -F[‘t’] ‘{if (NR &gt; 1) {if ($NF &gt; 0) {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”t”(-log($NF)/log(10))} else {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”t500”}}}’ <em>inp.bed</em> &gt; <em>out_WashU.bed</em></p>
</div></blockquote>
</div></blockquote>
</li>
<li><p class="first">Output of FitHiChIP(L+M) or FitHiChIP(S+M) -  file <strong>FitHiC_Q”qval”_MergeNearContacts.bed</strong></p>
<blockquote>
<div><p>Let us denote the file as <em>inp_merged.bed</em>.
It contains header line, first 6 fields denote the interacting bins and the 9th field stores the q-value.
User needs to apply the following awk script to this file:</p>
<p>awk -F[‘t’] ‘{if (NR &gt; 1) {if ($9 &gt; 0) {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”t”(-log($9)/log(10))} else {print $1”,”(($2+$3)/2-1)”,”(($2+$3)/2+1)”t”$4”,”(($5+$6)/2-1)”,”(($5+$6)/2+1)”t500”}}}’ <em>inp_merged.bed</em> &gt; <em>out_merged_WashU.bed</em></p>
</div></blockquote>
</li>
</ol>
</div></blockquote>
<p>Now user needs to upload the generated .bed file (<strong>Note: we do not need to convert them in gzipped / tabix formats</strong>) in the epigenome browser, by
performing the following steps.</p>
<blockquote>
<div><ol class="arabic simple">
<li>Open &lt;<a class="reference external" href="http://epigenomegateway.wustl.edu/legacy/">http://epigenomegateway.wustl.edu/legacy/</a>&gt;</li>
<li>Select reference genome.</li>
<li>Select <strong>Genome Browser</strong> -&gt; <strong>Apps</strong> -&gt; <strong>File upload</strong> -&gt; <strong>Choose files</strong></li>
<li>Select the generated WashU bed file and click <strong>setup</strong></li>
<li>Select Radio button for <strong>Pairwise interaction</strong> -&gt; <strong>add as Track</strong></li>
<li>Now right click on the track in the browser, and select <strong>Arc</strong>.</li>
</ol>
</div></blockquote>
</div>
</div>
<div class="section" id="visualizing-significant-interactions-in-ucsc-genome-browser">
<h2>Visualizing significant interactions in UCSC genome browser<a class="headerlink" href="#visualizing-significant-interactions-in-ucsc-genome-browser" title="Permalink to this headline">¶</a></h2>
<p>FitHiChIP loops can be converted to the bigBed format (or precisely BigInteract format) for visualizing in UCSC genome browser.
User should perform the following steps:</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">Check the link &lt;<a class="reference external" href="https://genome.ucsc.edu/goldenPath/help/interact.html">https://genome.ucsc.edu/goldenPath/help/interact.html</a>&gt;, example 4, for reference.</p>
</li>
<li><p class="first">Download UCSC <strong>bedToBigBed</strong> utility function, store it in system.</p>
</li>
<li><p class="first">Download the file <strong>interact.as</strong> (specified in the above mentioned link) and store it in the system.</p>
</li>
<li><p class="first">Then, to convert the FitHiChIP(L) or FitHiChIP(S) generated significant interactions (file <strong>FitHiC_Q”qval”.bed</strong>) into bed format, using the following command:</p>
<blockquote>
<div><p>awk -F’[t]’ ‘{if (NR&gt;1) {if ($NF&gt;0) {a=int(-log($NF)/log(10)) + 1} else {a=500}; b=100; print $1”t”(($2+$3)/2-1)”t”(($5+$6)/2+1)”t.t”a”t”b”t.t#0000FFt”$1”t”(($2+$3)/2-1)”t”(($2+$3)/2+1)”t.t.t”$4”t”(($5+$6)/2-1)”t”(($5+$6)/2+1)”t.t.”}}’ <em>FitHiC_Q”qval”.bed</em> | sort -k1,1 -k2,2n -k3,3n &gt; <em>out_FitHiChIP_converted.bed</em></p>
</div></blockquote>
</li>
<li><p class="first">Then convert the generated file into BigBed format as</p>
<blockquote>
<div><p>/path/to/bedToBigBed -as=/path/to/interact.as -type=bed5+13 <em>out_FitHiChIP_converted.bed</em> <em>ref_genome_chr.sizes</em> <em>out_loop.bb</em></p>
<p>where,</p>
<blockquote>
<div><ol class="arabic simple">
<li>/path/to/bedToBigBed: path of <strong>bedToBigBed</strong> utility in local system</li>
<li>/path/to/interact.as: path of the file <strong>interact.as</strong> in local system</li>
<li><em>ref_genome_chr.sizes</em>: Same as the parameter (B.4) (ChrSizeFile) in FitHiChIP configuration options.</li>
<li>The file <strong>out_loop.bb</strong> is the BigBed converted significant interactions. User can upload this file in UCSC genome browser for visualization.</li>
</ol>
</div></blockquote>
</div></blockquote>
</li>
</ol>
<p>6. To convert the output of FitHiChIP(L+M) or FitHiChIP(S+M) (file <strong>FitHiC_Q”qval”_MergeNearContacts.bed</strong>), replace “$NF” with “$9” in the command
mentioned in (4). Rest of the procedure is similar.</p>
</div></blockquote>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2019, Sourya Bhattacharyya, Ferhat Ay.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.3.<br/>
    </p>
  </div>
</footer>
  </body>
</html>